tests-beta/android/MASVS-PLATFORM/MASTG-TEST-02te.md

```diff
@@ -0,0 +1,32 @@
+---
+platform: android
+title: Testing WebViews Cleanup
+id: MASTG-TEST-02te // TODO allocate real ID
+type: [dynamic, manual]
+weakness: MASWE-0118
+profiles: [L1, L2]
+best-practices: [MASTG-BEST-00be]
+prerequisites:
+- identify-sensitive-data
+---
+
+## Overview
+
+This test verifies that the application cleans up sensitive information used by WebViews.
+
+@MASTG-KNOW-0018 describes the different storage areas used by WebViews.
+
+## Steps
+
+1. Install and run the app.
+2. Navigate to the webview of the mobile app you want to ensure is appropriately cleaned up.
+3. Optionally, manually debug the webview by @MASTG-TECH-0141 to inspect the storage areas for their content.
+4. Execute a method trace (@MASTG-TECH-0033) (using e.g. @MASTG-TOOL-0001) by attaching to the running app, targeting webview cleanup APIs, and save the output.
+
+## Observation
+
+The output should list the locations in the app where webview cleanup APIs are used during the current execution.
```

Comment:
I think "the lack of them" is very interesting in this case. The test should account for this. Are we seeing `setDomStorageEnabled`, `setAppCacheEnabled` or `setDatabaseEnabled` but not their "clearing" methods? Is the app using `localStorage.setItem` and not clearing?

Also, if an app doesn't even use WebViews this test shouldn't fail. It would not apply. If the WebView doesn't handle sensitive data the test shouldn't fail either.

```suggestion
The output should list the locations in the app where WebView cleanup APIs are used during the current execution or the lack of them.
```